MapboxNavigation/NavigationMapView.swift

```diff
@@ -1350,3 +1389,130 @@ extension NavigationMapView {
+    public func unhighlightBuildings() {
+        highlightBuildings(nil, in3D: false, extrudeAll: false)
+    }
+    
+    private func highlightBuildings(with attributes: [BuildingHighlightAttributes], in3D: Bool) {
+        let buildingAttributes = attributes.map {(value) -> BuildingHighlightAttributes in
+            var output = value
+            if output.identifier == nil {
+                output.identifier = buildingId(coordinate: value.location)
+            }
+            return output
+        }
+        let buildingsToHighlight = buildingAttributes.filter { $0.identifier != nil }.map {($0.identifier!, $0.highlightColor)}
+        if buildingsToHighlight.count > 0 {
+            highlightBuildings(buildingsToHighlight, in3D: in3D, extrudeAll: false)
+        } else {
+            unhighlightBuildings()
+        }
+    }
+    
+    private func prepareBuildingsForQuerying() {
+        guard highlightedBuildingLayer == nil else { return }
+        
+        if let buildingsSource = style?.source(withIdentifier: "composite") {
+            highlightedBuildingLayer = MGLFillExtrusionStyleLayer(identifier: StyleLayerIdentifier.buildingExtrusion, source: buildingsSource)
+            highlightedBuildingLayer?.sourceLayerIdentifier = "building"
+            highlightedBuildingLayer?.fillExtrusionColor = NSExpression(forConstantValue: buildingColor)
+            highlightedBuildingLayer?.fillExtrusionOpacity = NSExpression(forConstantValue: 0.05)
+            highlightedBuildingLayer?.fillExtrusionHeightTransition = MGLTransition(duration: 0.8, delay: 0)
+            highlightedBuildingLayer?.fillExtrusionOpacityTransition = MGLTransition(duration: 0.8, delay: 0)
+  
+            if let highlightedBuildingLayer = highlightedBuildingLayer {
+                style?.addLayer(highlightedBuildingLayer)
+            }
+        }
+    }
+    
+    private func removeBuildingsLayer() {
+        if let highlightedBuildingLayer = highlightedBuildingLayer {
+            style?.removeLayer(highlightedBuildingLayer)
+            self.highlightedBuildingLayer = nil
+        }
+    }
+
+    private func buildingId(coordinate: CLLocationCoordinate2D) -> UInt64? {
+        let screenCoordinate = convert(coordinate, toPointTo: self)
+
+        // To increase a chance of selecting correct building ID filter out
+        // features which do not contain appropriate attributes.
+        let features = visibleFeatures(at: screenCoordinate).filter {
+            let extrude = $0.attribute(forKey: "extrude") as? String
+            let type = $0.attribute(forKey: "type") as? String
+            return extrude == "true" && type == "building"
+        }
+        
+        if let feature = features.first, let buildingId = feature.identifier as? UInt64 {
+            return buildingId
+        }
+        
+        return nil
+    }
+    
+    private func highlightBuildings(_ buildings: [(identifier: UInt64, highlightColor: UIColor)]?, in3D: Bool, extrudeAll: Bool) {
```

Comment:
Since `extrudeAll` feature is not really supported I propose to remove it completely. If needed we can add it in future, but for now it'll simplify current logic.

Reply:
I'd like to keep this capability in for further design and UX experimentation

Reply:
Does the Android implementation have the ability to highlight multiple buildings in different colors at the same time? If not, we should ticket that out as tail work in the mapbox/mapbox-navigation-android repository.

Reply:
I think @d-prukop can provide more info on this one.